CMakeLists.txt

```diff
@@ -25,15 +25,119 @@ set(PROJECT_VERSION_PATCH 0)
 set(BUILD_SHARED_LIBS ON)
 ## set our cmake module path
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+include(CMakeDependentOption)
+include(CheckIncludeFile)
+include(ExternalProject)
+include(DownloadProject)
+include(ExecuteMaven)
+include(CheckCXXCompilerFlag)
+
+
+option(DOWNLOAD_DEPENDENCIES "Downloads and builds all dependencies locally " OFF)
+option(HBASE_TARGET_TAG "HBase tag to be used if HBASE_HOME is not set" "e5345b3a7c32c6a80394319c17540b84c8fe66ba")
```

Comment:
Since https://github.com/apache/hbase/blob/master/hbase-common/src/saveVersion.sh#L23 wasn't updated until 13 days ago I chose master at the time. I didn't want to use the master branch since that could change but what is built via maven is a relatively small set of low risk changing items. I'll admit I don't have a strong preference and will change it to master and add an item to change to a tag in the future upon the next release.